app/code/Magento/Indexer/Model/Indexer.php

```diff
@@ -418,6 +418,11 @@ public function reindexAll()
                 $state->save();
                 $this->getView()->resume();
                 throw $exception;
+            }catch (\Error $error) {
```

Comment:
- please add space before catch
- is there more specific exception you can catch?

Reply:
It not so easy to a more specific exception:
Currently,  **php7** throw much different but you must catch all to set the correct state for the indexer. After a fatal, the indexer must be on state **invalid**

<img width="484" alt="php__errors_in_php_7_-_manual" src="https://user-images.githubusercontent.com/5289370/31093975-c59f24dc-a7b3-11e7-845f-e48f1de0f9b8.png">